install/helm.go

```diff
@@ -113,8 +113,6 @@ func (k *K8sInstaller) getHelmValues() (map[string]interface{}, error) {
 			}
 		}
 
-		// TODO: remove when removing "cluster-name" flag (marked as deprecated),
-		// kept for backwards compatibility
 		if k.params.ClusterName != "" {
 			helmMapOpts["cluster.name"] = k.params.ClusterName
 		}
```

Comment:
we can't remove this block because we need to propagate the cluster name that gets auto-detected in https://github.com/cilium/cilium-cli/blob/main/install/autodetect.go#L150